src/wolf.system/cmake/external.cmake

```diff
@@ -0,0 +1,240 @@
+			target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/debug/libboost_atomic.a)
+            target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/debug/libboost_context.a)
+            target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/debug/libboost_fiber.a)
+            target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/debug/libboost_date_time.a)
+            target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/debug/libboost_regex.a)
+        else()
+			target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/release/libboost_atomic.a)
+            target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/release/libboost_context.a)
+            target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/release/libboost_fiber.a)
+			target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/release/libboost_date_time.a)
+			target_link_libraries(Boost INTERFACE ${WOLF_EXTERN_DEPS_DIR}/boost/lib/linux/x64/release/libboost_regex.a)
+        endif()
+    endif()
+endfunction()
+
+function(wolf_find_quiche)
+    message(STATUS "wolf dependency quiche")
+
+    list(APPEND HANDLED_BY_EXTERNAL quiche)
+
+    add_library(quiche INTERFACE)
+
+    target_include_directories(quiche INTERFACE "${WOLF_EXTERN_DEPS_DIR}/quiche/include")
+
+    target_link_libraries(quiche INTERFACE "${WOLF_EXTERN_DEPS_DIR}/quiche/lib/${PREFIX_LIB_PATH}/${CMAKE_BUILD_TYPE}/quiche.lib")
+endfunction()
+
+function(wolf_find_activemq_cpp)
+    if (NOT UNIX)
+        message(FATAL_ERROR "wolf dependency active-mq: it's only available for *nix systems.")
+    endif()
+
+    message(STATUS "wolf dependency activemq-cpp")
+
+    FetchContent_Declare(
+        activemq_cpp
+        GIT_REPOSITORY https://github.com/apache/activemq-cpp
+        GIT_TAG        master
+    )
+
+    list(APPEND HANDLED_BY_EXTERNAL activemq_cpp)
+
+    FetchContent_GetProperties(activemq_cpp)
+    if (NOT activemq_cpp_POPULATED)
+        FetchContent_Populate(activemq_cpp)
+
+        message(STATUS "building activemq-cpp at ${activemq_cpp_BINARY_DIR}")
+
+        find_program(MAKE_PROGRAM make REQUIRED)
+
+        add_custom_target(
+            activemq-cpp-build
+            COMMAND chmod +x ./autogen.sh
+            COMMAND ./autogen.sh
+            COMMAND ./configure --prefix=${activemq_cpp_BINARY_DIR}
+            COMMAND ${MAKE_PROGRAM} install
+            WORKING_DIRECTORY "${activemq_cpp_SOURCE_DIR}/activemq-cpp"
+            BYPRODUCTS "${activemq_cpp_BINARY_DIR}"
+        )
+
+        add_library(activemq-cpp INTERFACE)
+        add_library(activemq-cpp::activemq-cpp ALIAS activemq-cpp)
+
+        target_include_directories(activemq-cpp INTERFACE
+            ${activemq_cpp_BINARY_DIR}/include/activemq-3.10.0/)
```

Comment:
Library `ActiveMQ` is set to fetch always the `master` branch, however, in this line the include folder with the specific name (version) is added to the project. This may cause failure in building the project in the future if `ActiveMQ` is updated. Please consider another approach for including the header files of `ActiveMQ`.

Reply:
it's a requirement of `activemq-cpp` itself and the behavior is preserved from previous version of this file:

https://github.com/WolfEngine/WolfEngine/blob/wolf-2/src/wolf.system/CMakeLists.txt#L457 ...